Support mac OS binary release for CI

peanut996 opened the pull request:
### Description

#### Summary
This PR adds macOS platform support to our CI. It modifies key files for smooth macOS builds and releases.

#### Changes Made
- **`.github/workflows/publish.yml`**:
    - Added macOS entries to the `os` matrix.
    - Adjusted JDK download, caching, and Gradlew steps for macOS.
- **`buildSrc/src/main/kotlin/buildlogic/CiUtils.kt`**: Updated functions to handle macOS architecture for file naming.
- **`desktop/app/build.gradle.kts`**: Added code to work with macOS architecture in packaging.

### Example 
<img width="1240" alt="image" src="https://github.com/user-attachments/assets/1c6905c3-4fe5-4693-8504-b27211328ae6">

### Test locally
<img width="1068" alt="image" src="https://github.com/user-attachments/assets/6d185350-e826-4dc3-b992-d0494b4c0883">

peanut996 commented:
But the window button seems conflict.

I'll solve this later.

amir1376 commented:
Thanks for you interest to contributing to this project but unfortunately I can't merge it for now.
i intentionally not added mac target, as I mentioned [here](https://github.com/amir1376/ab-download-manager/issues/16) I don't have a Mac device to test and debug the app. if I add or modify the structure of the app it may breaks in the mac user systems. 
and it is not that I want for now!

yinglian-litao commented:
> ### Description
> #### Summary
> This PR adds macOS platform support to our CI. It modifies key files for smooth macOS builds and releases.
> 
> #### Changes Made
> * **`.github/workflows/publish.yml`**:
>   
>   * Added macOS entries to the `os` matrix.
>   * Adjusted JDK download, caching, and Gradlew steps for macOS.
> * **`buildSrc/src/main/kotlin/buildlogic/CiUtils.kt`**: Updated functions to handle macOS architecture for file naming.
> * **`desktop/app/build.gradle.kts`**: Added code to work with macOS architecture in packaging.
> 
> ### Example
> <img alt="image" width="1240" src="https://private-user-images.githubusercontent.com/32303768/386514818-1c6905c3-4fe5-4693-8504-b27211328ae6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMxNTAxNjQsIm5iZiI6MTczMzE0OTg2NCwicGF0aCI6Ii8zMjMwMzc2OC8zODY1MTQ4MTgtMWM2OTA1YzMtNGZlNS00NjkzLTg1MDQtYjI3MjExMzI4YWU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjAyVDE0MzEwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlYTYwMjFjODQ4YTcwYzZkZDBmODM1NmZlMjU3YjVlNDYxMDhjNTczODBjNzhiYmI4ZGRkYWI2YzZlZTYzOTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.S-Knen-WmLFBIcS-_y0ENnRVbBQH8NQm5Kadwy1NCiU">
> ### Test locally
> <img alt="image" width="1068" src="https://private-user-images.githubusercontent.com/32303768/386334038-6d185350-e826-4dc3-b992-d0494b4c0883.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMxNTAxNjQsIm5iZiI6MTczMzE0OTg2NCwicGF0aCI6Ii8zMjMwMzc2OC8zODYzMzQwMzgtNmQxODUzNTAtZTgyNi00ZGMzLWI5OTItZDA0OTRiNGMwODgzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjAyVDE0MzEwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNmZjAzYzUxZDMzYTU2MjQ2NWYzNmNhYjkyYjcwZGY0MWQyNGUzOGIxN2UzZjJlMDlhNTk4OWZmZDhiNWU4NTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.qhtJa6iNZFmSPUnwxBAmvXPg0i7Ec7MPty5BZtbCCMs">

兄弟，你测试没问题可以fork一份编译好的发出来。不是正式版本，我们也能先体验当测试版本开心下。

peanut996 commented:
> > ### Description
> > #### Summary
> > This PR adds macOS platform support to our CI. It modifies key files for smooth macOS builds and releases.
> > #### Changes Made
> > 
> > * **`.github/workflows/publish.yml`**:
> >   
> >   * Added macOS entries to the `os` matrix.
> >   * Adjusted JDK download, caching, and Gradlew steps for macOS.
> > * **`buildSrc/src/main/kotlin/buildlogic/CiUtils.kt`**: Updated functions to handle macOS architecture for file naming.
> > * **`desktop/app/build.gradle.kts`**: Added code to work with macOS architecture in packaging.
> > 
> > ### Example
> > <img alt="image" width="1240" src="https://private-user-images.githubusercontent.com/32303768/386514818-1c6905c3-4fe5-4693-8504-b27211328ae6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMxNTAxNjQsIm5iZiI6MTczMzE0OTg2NCwicGF0aCI6Ii8zMjMwMzc2OC8zODY1MTQ4MTgtMWM2OTA1YzMtNGZlNS00NjkzLTg1MDQtYjI3MjExMzI4YWU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjAyVDE0MzEwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlYTYwMjFjODQ4YTcwYzZkZDBmODM1NmZlMjU3YjVlNDYxMDhjNTczODBjNzhiYmI4ZGRkYWI2YzZlZTYzOTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.S-Knen-WmLFBIcS-_y0ENnRVbBQH8NQm5Kadwy1NCiU">
> > ### Test locally
> > <img alt="image" width="1068" src="https://private-user-images.githubusercontent.com/32303768/386334038-6d185350-e826-4dc3-b992-d0494b4c0883.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMxNTAxNjQsIm5iZiI6MTczMzE0OTg2NCwicGF0aCI6Ii8zMjMwMzc2OC8zODYzMzQwMzgtNmQxODUzNTAtZTgyNi00ZGMzLWI5OTItZDA0OTRiNGMwODgzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjAyVDE0MzEwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNmZjAzYzUxZDMzYTU2MjQ2NWYzNmNhYjkyYjcwZGY0MWQyNGUzOGIxN2UzZjJlMDlhNTk4OWZmZDhiNWU4NTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.qhtJa6iNZFmSPUnwxBAmvXPg0i7Ec7MPty5BZtbCCMs">
> 
> 兄弟，你测试没问题可以fork一份编译好的发出来。不是正式版本，我们也能先体验当测试版本开心下。

你可以在这里尝鲜MacOS版本，应该也都是能用的地步。

窗口什么还没修，不过只是尝鲜哈。

https://github.com/peanut996/ab-download-manager/releases/tag/v1.4.2
